docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,438 @@
+    // or negatively scored, returning a bool representing whether the given
+    // peer is protected or not.
+    MutualRm(id peer.ID, tag string) bool
+    // IsMutual returns whether the given peer is a mutual peer.
+    IsMutual(id peer.ID, tag string) bool
+  
+    // BandwidthStats returns a Stats struct with bandwidth metrics for all
+    // data sent/received by the local peer, regardless of protocol or remote 
+    // peer IDs.
+    BandwidthStats() Stats
+    // BandwidthForPeer returns a Stats struct with bandwidth metrics associated
+    // with the given peer.ID. The metrics returned include all traffic sent /
+    // received for the peer, regardless of protocol.
+    BandwidthForPeer(id peer.ID) Stats
+    // BandwidthForProtocol returns a Stats struct with bandwidth metrics 
+    // associated with the given protocol.ID.
+    BandwidthForProtocol(proto protocol.ID) Stats
+   
+    // ResourceState returns the state of the resource manager.
+    ResourceState() rcmgr.ResourceManagerStat
+   
+    // PubSubPeers returns the peer IDs of the peers joined on
+    // the given topic.
+    PubSubPeers(topic string) ([]peer.ID, error)
+  }
+```
+
+### NodeModule
+
+```go
+
+  type NodeModule interface {
+    // Type returns the node type.
+    Type() node.Type
+    // Version returns information about the current binary build.
+    Version() string
+   
+    // LogLevelSet sets the given component log level to the given level.
+    LogLevelSet(ctx context.Context, name, level string) error
+  }
+  
+```
+
+#### DAS
+
+```go
+  type DASModule interface {
+    // Stats returns current stats of the DASer.
+    Stats() (das.SamplingStats, error)
+  }
+```
+
+#### State
+
+```go
+  type StateModule interface {
+    // AccountAddress retrieves the address of the node's account/signer
+    AccountAddress(ctx context.Context) (state.Address, error)
+    // Balance retrieves the Celestia coin balance for the node's account/signer
+    // and verifies it against the corresponding block's AppHash.
+    Balance(ctx context.Context) (*state.Balance, error)
+    // BalanceForAddress retrieves the Celestia coin balance for the given 
+    // address and verifies the returned balance against the corresponding
+    // block's AppHash.
+    BalanceForAddress(ctx context.Context, addr state.Address) (*state.Balance, error)
+    // SubmitTx submits the given transaction/message to the Celestia network
+    // and blocks until the tx is included in a block.
+    SubmitTx(ctx context.Context, tx state.Tx) (*state.TxResponse, error)
+    // SubmitPayForData builds, signs and submits a PayForData transaction.
+    SubmitPayForData(
+      ctx context.Context, 
+      nID namespace.ID, 
+      data []byte, 
+      gasLimit uint64,
+    ) (*state.TxResponse, error)
+    // Transfer sends the given amount of coins from default wallet of the node 
+    // to the given account address.
+    Transfer(
```

Comment:
[non-blocking]
Do we want a full balance transfer option?  Or have we discussed how someone would do that?

I assume the full cost of this method is amount + gasFee. So it's hard for users to completely empty a wallet. 
But if we can structure it to allow for amount being the full cost, and what is sent to the new address is amount - gasFee, then users can drain a wallet easily. 

Something a simple as a `subGass bool` argument or just a second method.